src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int capacity;
+    private int size;
+    private int threshold;
+
+    private Node<K, V>[] container;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        this.container = (Node<K, V>[]) new Node[capacity];
+        this.threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = key == null ? 0 : getIndexUsingHashcode(key);
+
+        Node<K, V> currentNode = container[index];
+        while (currentNode != null) {
+            if (currentNode.key == key || currentNode.key != null && currentNode.key.equals(key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
 
+        if (size++ > threshold) {
+            increaseSize();
+        }
+
+        Node<K, V> newNode = new Node<>(key == null ? 0 : key.hashCode(), key, value, null);
+        if (container[index] == null) {
+            container[index] = newNode;
+        } else {
+            newNode.next = container[index].next;
+            container[index].next = newNode;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> lookingNode = container[key == null ? 0 : getIndexUsingHashcode(key)];
+        while (lookingNode != null) {
+            if (lookingNode.key == key || lookingNode.key != null && lookingNode.key.equals(key)) {
+                return lookingNode.value;
+            }
+            lookingNode = lookingNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
     }
+
+    private static class Node<K, V> {
+        private int hashCode;
```

Comment:
I think we don't need this field because you use it only when creating a Node.

Reply:
Actually it is used later not to calculate hashCode each time it is called but as I don't use it right now I will remove it